### PR TITLE
feat(route): add AtCoder

### LINKED ISSUE #8824: Wish to add atcoder programing upcoming contests list
### Category

Programming

### Website URL

https://atcoder.jp/contests/

### Website description

If you can help me add "Upcoming Contests" parts at https://atcoder.jp/contests/. 
That's will be much helpful.

### What content should be included?

If you can help me add "Upcoming Contests" parts at https://atcoder.jp/contests/. 
That's will be much helpful.

### Additional description

_No response_

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by nczitzk**
## 该 PR 相关 Issue / Involved issue

Close #8824

## 完整路由地址 / Example for the proposed route(s)

```routes
/atcoder/contest/en/upcoming
/atcoder/post
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [x] Documentation
  - [x] CN
  - [x] EN
- [x] 全文获取 fulltext
  - [x] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/atcoder/contest.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                link: `${rootUrl}${item.attr('href')}`,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
```
Comment (a180285) [inline lib/v2/atcoder/contest.js:45]: Thank you very much for help on my issue.

I found that the item description is still Japan, so I suggest to add language param for item detail.
You can check with your Preview link: https://rsshub-682rmkf7u-diy.vercel.app/atcoder/contest/en/upcoming

```suggestion
                    url: `${item.link}?lang=${language}`,
```

File: lib/v2/atcoder/contest.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+            return {
+                title: item.text(),
+                link: `${rootUrl}${item.attr('href')}`,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
```
Comment (a180285) [inline lib/v2/atcoder/contest.js:42]: En, this link should be changed, too.

File: lib/v2/atcoder/contest.js (hunk 1 above)
Comment (nczitzk) [reply lib/v2/atcoder/contest.js:45]: Thanks for the review 😊

Comment (a180285) [reply lib/v2/atcoder/contest.js:45]: But why the output is still Japanese? Interesting, maybe a bug:

https://rsshub-1ei9fpduf-diy.vercel.app/atcoder/contest/en/upcoming

**Comment by a180285** [comment]
Thanks, LGTM

**Pull request merged**